### PR TITLE
[terra-clinical-onset-picker] Announce state and chosen/selected item in the combo box

### PR DESCRIPTION
**Pull request by vinaybhargavar**
### Summary
The state and chosen/selected item in the combo box must be announced by the screen reader. Currently, we set the aria-labelledby to the id of the label of the combo box. Removing the aria-labelledby announces both the state (expanded/collapsed) and the chosen/selected item in the combo box. This can be seen visually but the screen reader does not call this out. The date format was not read correctly by the screen reader. The screen reader was reading Date Date Date/Date/Date. We will need to hide the label for the screen reader users.

**What was changed:**
Removed the aria-labelledby from all the fields. Hid the label for Date Field for screen reader users by setting the aria-hidden attribute.

**Why it was changed:**
For the screen reader to call both the state and chosen item in the combo box.

### Testing

This change was tested using:

- [ ] WDIO
- [ ] Jest
- [x] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

On MAC:

State & Chosen item:
https://github.com/cerner/terra-clinical/assets/55503680/d3dd35ab-d700-4703-aeb0-af3edae7864c

Date format:

https://github.com/cerner/terra-clinical/assets/55503680/d343d997-675d-41de-b89a-a5278f0bdb47

https://github.com/cerner/terra-clinical/assets/55503680/fd0da416-a0e6-40ec-b0ca-8893455b9d02

On Windows:

https://github.com/cerner/terra-clinical/assets/55503680/ecf987c8-4138-43ac-9b1f-8630f2540433

Date Format:

https://github.com/cerner/terra-clinical/assets/55503680/d74557d6-ed0d-4e24-9e51-4ffcc1913fa8

Deployed site - https://engineering.cerner.com/terra-clinical/pull/896/

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [x] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-9428

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

**Comment by vinaybhargavar** [comment]
Deployed site - https://engineering.cerner.com/terra-clinical/pull/896/

**Review by chenyixuan625 — COMMENTED** [review]
Code change looks good to me. Could we upload a video of the updated behavior of reading the date format as well?

File: packages/terra-clinical-onset-picker/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Updated
+  * Updated `[terra-clinical-onset-picker]` to announce state and chosen/selected item in the combo box
```
Comment (RayGunY) [inline packages/terra-clinical-onset-picker/CHANGELOG.md:6]: nit: add a period

**Review by chenyixuan625 — APPROVED** [review]

**Review by RayGunY — APPROVED** [review]

**Comment by vinaybhargavar** [comment]
@sdadn, @sycombs - This PR is ready for your review. TIA.

**Review by sycombs — APPROVED** [review]

**Review by scottwilmarth — APPROVED** [review]

**Review by sdadn — APPROVED** [review]

**Pull request merged**